src/Engine/System/ParticleSystem.cpp

```diff
@@ -0,0 +1,15 @@
+#include "ParticleSystem.hpp"
+
+const Vector<ParticleComponent *> & ParticleSystem::s_particleComponents(Scene::getComponents<ParticleComponent>());
+void ParticleSystem::init() {
+
+}
+
+void ParticleSystem::update(float dt) {
+    for (auto &p : s_particleComponents) {
+        p->update(dt);
+        if (p->Dead()) {
+            Scene::destroyGameObject(p->gameObject());
```

Comment:
We destroy the entire game object when a particle component dies?

Reply:
Yeah... I'll just remove the component.

Reply:
heh